### PR TITLE
Make the language check a part of CanSendHail

### PR DESCRIPTION
**Pull request by Amazinite**
**Bugfix:** This PR might address issue #4078.

## Fix Details
Adds the check to see if the hailing ship shares a language with the player to the CanSendHail function. This should prevent ships from hailing the player when a language is not shared under any circumstance, as CanSendHail is always checked before sending a hail message.

Still not sure how the World Ship in #4078 sent a disabled hail, though, as disabled ships do not send random hails, and the only way to view disabled hails is to hail a disabled ship, as best I can tell.

## Testing Done
Flew around Ka'het space with and without this change. Without this change, Ka'het ships would occasionally hail me, but hailing them returned the expected alien language response. With this change, I was never hailed by a Ka'het ship. The same should extend to Korath ships.

## Save File
N/A

### CODE REVIEW

File: source/Engine.cpp [hunk 1]
```diff
@@ -139,6 +140,11 @@ namespace {
 				|| ship->Cloaking() >= 1. || ship->GetPersonality().IsMute())
 			return false;
 		
+		// Ships that don't share a language with the player shouldn't send hails.
+		const Government *gov = ship->GetGovernment();
```
Comment (petervdmeer) [inline source/Engine.cpp:144]: Is it possible that a ship has no government (i.e. you getting a null-pointer here)?

**Review by petervdmeer — APPROVED** [review]
One remark (not sure if it is relevant though).
Change looks good otherwise.

File: source/Engine.cpp (hunk 1 above)
Comment (Amazinite) [reply source/Engine.cpp:144]: A nullptr check is done further up in this function. (Moving this line up above that one as I hadn't noticed it earlier.)

**Comment by Sandwichs-del** [comment]
I think it would be better to clearly distinguish the technological and language barrier.
I should receive nothing if it is due to a technical barrier, but I should receive "Ship XXX tries to hail you, but you do not understand what it says" when it is due to a language barrier.

Since I can hail them I see no reason why they shouldn't be able to hail me.
Would you accept a patch that overrides the message for such cases?

**Pull request merged**